File: skellig-test-step-processing/src/test/java/org/skellig/teststep/processing/utils/UnitTestUtils.java
package org.skellig.teststep.processing.utils;

import org.skellig.teststep.processing.model.ExpectedResult;

import java.util.HashMap;
import java.util.List;
import java.util.Map;

public final class UnitTestUtils {

    public static Map<String, Object> createMap(Object... params) {
        Map<String, Object> map = new HashMap<>();
        for (int i = 0; i < params.length; i += 2) {
            map.put((String) params[i], params[i + 1]);
        }
        return map;
    }

    public static ExpectedResult extractExpectedValue(ExpectedResult expectedResult, int... indexPath) {
        for (int index : indexPath) {
            expectedResult = expectedResult.<List<ExpectedResult>>getExpectedResult().get(index);
        }
        return expectedResult;
    }
}

File: skellig-test-step-processing/src/test/kotlin/org/skellig/teststep/processing/utils/UnitTestUtils.kt
package org.skellig.teststep.processing.utils

import org.skellig.teststep.processing.model.ExpectedResult

sealed class UnitTestUtils {

    companion object {
        fun createMap(vararg params: Any?): Map<String, Any?> {
            val map: MutableMap<String, Any?> = HashMap()
            var i = 0
            while (i < params.size) {
                map[params[i] as String] = params[i + 1]
                i += 2
            }
            return map
        }

        fun extractExpectedValue(expectedResult: ExpectedResult, vararg indexPath: Int): ExpectedResult {
            var newExpectedResult = expectedResult
            for (index in indexPath) {
                newExpectedResult = newExpectedResult.get<List<ExpectedResult>>()!![index]
            }
            return newExpectedResult
        }
    }
}